### ISSUE TITLE
Early Stopping

### ISSUE THREAD

**Issue by JohannesMaxWel** [uclmr/jack#178]
I would be great to have some sort of early stopping functionality. There's two ways I see for how we can go about this, and both of which wouldn't be too hard to create, given the hooks and savers we already have.

1. Option A) store model parameters in regular intervals during training, stop training once validation loss increases (with some patience), use the latest model as final model.
2. Option B) train all the way until the pre-specified number of epochs is reached, store the model parameters at _every_ checkpoint, after training read from the metrics log file which epoch had the best parameters and load them as final model.

I think the first option is more elegant. Any opinions?
